lib/pages/settings/view/request_card.dart

```diff
@@ -0,0 +1,207 @@
+import 'package:acs_upb_mobile/authentication/model/user.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/settings/model/request.dart';
+import 'package:acs_upb_mobile/pages/settings/service/admin_provider.dart';
+import 'package:acs_upb_mobile/widgets/button.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
+
+class RequestCard extends StatefulWidget {
+  const RequestCard({this.requestId});
+
+  @required
+  final String requestId;
+
+  @override
+  _RequestCardState createState() => _RequestCardState();
+}
+
+class _RequestCardState extends State<RequestCard>
+    with AutomaticKeepAliveClientMixin {
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    final adminProvider = Provider.of<AdminProvider>(context);
+
+    return FutureBuilder(
```

Comment:
This is a bit too convoluted, let's split it up into smaller parts.

First, let's have a `buildUserHeader` method that returns `FutureBuilder(future: fetchUserById(...), ...` with a Column with the user name and group. When the data isn't loaded, either show empty strings or placeholders like "unknown" (check what looks nicer).

Another method can be `buildAcceptedMarker(bool accepted)` or `buildAcceptedTag` which returns the correctly-coloured widget. You can also remove a lot of the duplicate code here, this would look something like:
```dart
Widget buildAcceptedMarker(bool accepted) =>
  Container(
    padding: const EdgeInsets.all(5),
    decoration: BoxDecoration(
      border: Border.all(color: accepted ? Colors.green : Colors.red, width: 2),
      borderRadius: const BorderRadius.all(Radius.circular(6)),
    ),
    child: Text(
      accepted ? 'Accepted' : 'Denied',  // Also localize these strings
      style: Theme.of(context)
          .textTheme
          .headline6
          .copyWith(fontSize: 10),
    ),
  );
```

Another method can be `buildButtons(bool processed)`, and see my comment on the provider that the revert button should just call provider.revert instead of doing its own logic to figure out whether it was accepted or denied.